doc/lsst.pipe.base/task-framework-overview.rst

```diff
@@ -0,0 +1,49 @@
+.. NOTE: this topic should be edited into the modernized topic-based documentation style.
+
+.. _task-framework-overview:
+
+##############################
+Overview of the task framework
+##############################
+
+:doc:`lsst.pipe.base <index>` provides a data processing pipeline infrastructure.
+Data processing is performed by **tasks**, which are instances of `lsst.pipe.base.Task` or `lsst.pipe.base.CmdLineTask`.
+Tasks perform a wide range of data processing operations, from basic operations such as assembling raw images into CCD images (trimming overscan), fitting a WCS or detecting sources on an image, to complex combinations of such operations.
+
+**Tasks** are hierarchical.
+Each task may may call other tasks to perform some of its data processing; we say that a **parent task** calls a **subtask**.
+The highest-level task is called the **top-level task**.
+To call a subtask, the parent task constructs the subtask and then calls methods on it.
+Thus data transfer between tasks is simply a matter of passing the data as arguments to function calls.
+
+**Command-line tasks** are tasks that can be run from the command line.
+You might think of them as the LSST equivalent of a data processing pipeline.
+Despite their extra capabilities, command-line tasks can also be used as ordinary tasks and called as subtasks by other tasks.
```

Comment:
Not quite.  They are tasks that can _also_ be run from the command line.  You sort-of say this, but this statement is stronger -- they are tasks first and foremost.